### ISSUE TITLE
A “nondestructive” or modifying serializer

### ISSUE THREAD

**Issue by ModProg** [khonsulabs/rsn#8]
I was experimenting with this idea for TOML (https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=dbda15adfbdf58defcaf2c5707a3a033).
While obviously worse from a performance standpoint, I think having a `Serializer` that preserves whitespace and comments if possible and only updates modified values could be quite neat.

**Comment by ecton** [comment]
> could be quite neat

Yes, yes it could. I delegate this task to you though, because I really don't want to try to implement it :)

**Comment by ModProg** [comment]
No worries, I'll look into it :D when we have a value type that can lossless describe an rsn file, this should be very doable.
